pandas/core/ops/mask_ops.py

```diff
@@ -0,0 +1,178 @@
+"""
+Ops for masked ararys.
+"""
+from typing import Optional, Union
+
+import numpy as np
+
+from pandas._libs import lib, missing as libmissing
+
+
+def kleene_or(
+    left: Union[bool, np.ndarray],
+    right: Union[bool, np.ndarray],
+    left_mask: Optional[np.ndarray],
+    right_mask: Optional[np.ndarray],
+):
+    """
+    Boolean ``or`` using Kleene logic.
+
+    Values are NA where we have ``NA | NA`` or ``NA | False``.
+    ``NA | True`` is considered True.
+
+    Parameters
+    ----------
+    left, right : ndarray, NA, or bool
+        The values of the array.
+    left_mask, right_mask : ndarray, optional
+        The masks. Only one of these may be None, which implies that
+        the associated `left` or `right` value is a scalar.
+
+    Returns
+    -------
+    result, mask: ndarray[bool]
+        The result of the logical or, and the new mask.
+    """
+    # To reduce the number of cases, we ensure that `left` & `left_mask`
+    # always come from an array, not a scalar. This is safe, since because
+    # A | B == B | A
+    if left_mask is None:
+        return kleene_or(right, left, right_mask, left_mask)
+
+    assert isinstance(left, np.ndarray)
+
+    raise_for_nan(right, method="or")
+
+    if right is libmissing.NA:
+        result = left.copy()
```

Comment:
why do you need to copy?

Reply:
This avoids something like

```python
>>> a = pd.array([True, False, None])
>>> b = a | pd.NA
>>> b[0] = False
```

mutating `a` inplace. We have a test for this `TestLogicalOperators.test_scalar`.